src/scripts/selectors/UploaderSelectors.js

```diff
@@ -77,15 +77,32 @@ export const getTotalProgress: (state: RootState) => ?Number = createSelector(
     if (!video) {
       return 0
     }
-    // const uploadProgress = parseInt(video.uploadStatus.data.progress)
-    const uploadProgress = 0
-    // const transcodingProgress = parseInt(video.transcoderStatus.data.progress)
+    const uploadProgress =
+      parseInt(video.getIn(['uploadStatus', 'data', 'progress'])) || 0
     const transcodingProgress =
       parseInt(video.getIn(['transcodingStatus', 'data', 'progress'])) || 0
     return Math.floor((uploadProgress + transcodingProgress) / 2) || 0
   }
 )
 
+export const isUploaded = createSelector(
+  [getSelectedUploaderVideo],
+  upload => upload && upload.getIn(['uploadStatus', 'name']) === 'success'
+)
+
+export const isPublished = createSelector(
+  [getSelectedUploaderVideo],
+  upload => upload && (upload.published === true || upload.published === 'true')
+)
+
+export const isPublishable = createSelector(
+  [getSelectedUploaderVideo],
+  upload =>
+    upload &&
+    upload.getIn(['transcodingStatus', 'name']) === 'success' &&
+    upload.isPublished !== true
+)
```

Comment:
This totally works, but is a good opportunity to compose your selectors. You can use `isPublished` to help with `isPublishable` so that you do not duplicate any logic.

Example:

```javascript
export const isPublishable = createSelector(
  [getSelectedUploaderVideo, isPublished],
  (upload, uploadIsPublished) => upload && upload.getIn(['transcodingStatus', 'name']) === 'success' && !uploadIsPublished
)
```